packages/atomic/src/components/insight/atomic-insight-result-action/atomic-insight-result-action.tsx

```diff
@@ -16,25 +23,44 @@ import {Bindings} from '../../search/atomic-search-interface/atomic-search-inter
 })
 export class AtomicInsightResultAction implements InitializableComponent {
   @InitializeBindings() public bindings!: Bindings;
+  @ResultContext() private result!: Result;
   @State() public error!: Error;
 
+  @Event({
+    eventName: 'AtomicInsightResultActionClicked',
+    composed: true,
+    cancelable: true,
+    bubbles: true,
+  })
+  private actionClicked!: EventEmitter<InsightResultActionClickedEvent>;
+
   /**
    * Specify the result action icon to display.
    */
   @Prop({mutable: true}) public icon = Quickview;
 
   /**
-   * The text tooltip to show on the result action icon
+   * The text tooltip to show on the result action icon.
    */
   @Prop({mutable: true}) public tooltip = '';
 
+  /**
+   * The type of action to perform when the result action is clicked. This will be sent along the event fired when the button is clicked.
+   */
+  @Prop({mutable: true}) public action = '';
```

Comment:
Should this be typed in any way to stricten or guide what type of actions should be available, or do we really want `any string` ?

Reply:
You mean adding an Enum of possible actions? I would leave it to any since it just re-passes the action to events and you do whatever you want with it after anyway

Reply:
The actual accepted type should be reflected here: `public actions: Action`. Can also extend to `Action | string` if it's really the desired behaviour that it can be "anything" (I doubt it, but...)